api/incubator/src/main/java/io/opentelemetry/api/incubator/logs/ExtendedLogRecordBuilder.java

```diff
@@ -5,13 +5,124 @@
 
 package io.opentelemetry.api.incubator.logs;
 
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.Value;
+import io.opentelemetry.api.incubator.common.ExtendedAttributeKey;
+import io.opentelemetry.api.incubator.common.ExtendedAttributes;
 import io.opentelemetry.api.logs.LogRecordBuilder;
+import io.opentelemetry.api.logs.Severity;
+import io.opentelemetry.context.Context;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
 
 /** Extended {@link LogRecordBuilder} with experimental APIs. */
 public interface ExtendedLogRecordBuilder extends LogRecordBuilder {
 
   // keep this class even if it is empty, since experimental methods may be added in the future.
 
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setTimestamp(long timestamp, TimeUnit unit);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setTimestamp(Instant instant);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setObservedTimestamp(long timestamp, TimeUnit unit);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setObservedTimestamp(Instant instant);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setContext(Context context);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setSeverity(Severity severity);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setSeverityText(String severityText);
+
+  /** {@inheritDoc} */
+  @Override
+  ExtendedLogRecordBuilder setBody(String body);
+
+  /** {@inheritDoc} */
+  @Override
+  default ExtendedLogRecordBuilder setBody(Value<?> body) {
+    setBody(body.asString());
+    return this;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>NOTE: all standard {@link AttributeKey}-value pairs can also be represented as {@link
+   * ExtendedAttributeKey}-value pairs, but not all {@link ExtendedAttributeKey}-value pairs can be
+   * represented as standard {@link AttributeKey}-value pairs. From the standpoint of the emitted
+   * log record, there is no difference between adding attributes using the standard or extended
+   * attribute APIs.
+   */
+  @SuppressWarnings("unchecked")
+  @Override
+  default ExtendedLogRecordBuilder setAllAttributes(Attributes attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return this;
+    }
```

Comment:
I know this exists in the non-extended version and you're just going for parity, but since I'm looking at it -- I thought this kind of optimization doesn't actually buy much in practice, and that the `forEach` is basically a short circuit that will get jit optimized anyway?

Reply:
> in practice, and that the forEach is basically a short circuit that will get jit optimized anyway?

Could be? Not sure if the compiler would be smart enough to reach into the `forEach` implementation [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/internal/ImmutableKeyValuePairs.java#L93-L95) and determine that `data.length == 0`. 

I don't feel strongly about this.